packages/providers/upload-local/lib/index.js

```diff
@@ -17,32 +20,36 @@ module.exports = {
       }
     };
 
-    const publicDir = strapi.dirs.public;
+    // Ensure uploads folder exists
+    const uploadPath = path.resolve(strapi.dirs.public, UPLOADS_FOLDER_NAME);
+    try {
```

Comment:
Same as above + I would verify this at startup in the bootstrap of the plugin ?

Reply:
In which plugin would you do that?